### PR TITLE
Add support for AMD GPUs

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

File: src/umpire/CMakeLists.txt [hunk 1]
```diff
@@ -59,6 +59,21 @@ set(UMPIRE_ENABLE_LOGGING ${ENABLE_LOGGING})
 set(UMPIRE_ENABLE_SLIC ${ENABLE_SLIC})
 set(UMPIRE_ENABLE_ASSERTS ${ENABLE_ASSERTS})
 set(UMPIRE_ENABLE_STATISTICS ${ENABLE_STATISTICS})
+set(UMPIRE_ENABLE_ROCM ${ENABLE_ROCM})
```
Comment (mcfadden8) [inline src/umpire/CMakeLists.txt]: Did a description of this new option get added to https://github.com/LLNL/Umpire/blob/develop/docs/sphinx/advanced_configuration.rst?

Comment (davidbeckingsale) [reply src/umpire/CMakeLists.txt]: Nope 😞 I'll do that!

**Comment by mcfadden8** [comment]
Other than my comment on documentation, this looks great!

**Comment by davidbeckingsale** [comment]
@mcfadden8 - I updated the docs.

**Pull request merged**